### PR TITLE
strict-reconcile, Add strict-reconcile test

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
CNAO performs strict reconciling to make sure that un-allowed day2
changes are not done on its components.
In order to ensure this functionality does not break, we want to add
a test on a arbitrary component object.
Added a test to make sure strict-reconcile feature is preserved.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: automation/check-patch.e2e-nmstate-functests.sh [hunk 1]
```diff
@@ -40,12 +40,13 @@ main() {
     cd ${TMP_COMPONENT_PATH}
 
     make test-e2e-handler \
-        E2E_TEST_TIMEOUT=${TIMEOUT} \
-        E2E_TEST_ARGS="-ginkgo.noColor --junit-output=$ARTIFACTS/junit.functest.xml" \
-        HANDLER_NAMESPACE=$NAMESPACE \
+        E2E_TEST_TIMEOUT=$TIMEOUT \
+        e2e_test_args="-noColor" \
+        E2E_TEST_SUITE_ARGS="--operator-namespace $NAMESPACE --junit-output=$ARTIFACTS/junit.functest.xml" \
```
Comment (phoracek) [inline automation/check-patch.e2e-nmstate-functests.sh:45]: This change seems unrelated. Please keep it in a separate commit.

File: test/e2e/workflow/deployment_test.go [hunk 2]
```diff
@@ -143,10 +149,10 @@ var _ = Describe("NetworkAddonsConfig", func() {
 				MultusComponent,
 			}
 			configSpec := cnao.NetworkAddonsConfigSpec{
-				LinuxBridge: &cnao.LinuxBridge{},
-				Multus:      &cnao.Multus{},
-				Ovs:         &cnao.Ovs{},
-				MacvtapCni:  &cnao.MacvtapCni{},
+				LinuxBridge:            &cnao.LinuxBridge{},
```
Comment (phoracek) [inline test/e2e/workflow/deployment_test.go:152]: Unrelated to this PR. Please commit format fixes in a separate commit.

File: test/e2e/workflow/deployment_test.go [hunk 3]
```diff
@@ -249,6 +255,46 @@ var _ = Describe("NetworkAddonsConfig", func() {
...
 	})
+	Context("when Macvtap is deployed", func() {
+		BeforeEach(func() {
+			configSpec := cnao.NetworkAddonsConfigSpec{
+				MacvtapCni: &cnao.MacvtapCni{},
+			}
+			CreateConfig(gvk, configSpec)
+			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
+		})
+		FContext("and a forbidden day2 change is introduced to Macvtap daemonSet", func() {
```
Comment (phoracek) [inline test/e2e/workflow/deployment_test.go:266]: F

**Review by phoracek — CHANGES_REQUESTED** [review]

File: automation/check-patch.e2e-nmstate-functests.sh (hunk 1 above)
Comment (RamLavi) [reply automation/check-patch.e2e-nmstate-functests.sh]: Done

File: test/e2e/workflow/deployment_test.go (hunk 2 above)
Comment (RamLavi) [reply test/e2e/workflow/deployment_test.go]: Done

File: test/e2e/workflow/deployment_test.go (hunk 3 above)
Comment (RamLavi) [reply test/e2e/workflow/deployment_test.go]: Done

**Comment by RamLavi** [comment]
@phoracek fixes [here](https://github.com/kubevirt/cluster-network-addons-operator/compare/d67a8c061a45ded3869e737cc4bc542cbb110419..80e03bb339e4568618a6d72f82d629e6d7fb6cd3)

**Review by phoracek — APPROVED** [review]
/lgtm
/approve
